### DISCUSSION TITLE
Character_Class_Combination

### DISCUSSION THREAD

**Discussion by ghramsey** [MattFiler/LegendPlugin#96] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/51859157/209820362-6ae3f747-a577-4cf4-99c6-8e4fbd0cbc12.png)
Just curious how does this value of 132 derived?   There seem to be 20 possible character classes listed in that table.
Each assigned a number 0-19 correct?  Alien_Only seems to be on 2 since I see other filters that look to be for vents 
only the alien can navigate with character_class_combination value of 2.  So what combination gets to 132?  I'm not finding it.

**Comment by MattFiler** [comment, marked as answer]
It's a combination of multiple enum values (E.G. `ALIENS` and `ANDROIDS`), so won't be a regular index. Check this page out to understand the technical details: https://www.alanzucconi.com/2015/07/26/enum-flags-and-bitwise-operators/

I'm still unsure on the actual enum values - their order in that list on the wiki is correct, but I don't know if they all correspond to incrementing indexes (0,1,2,3,4,etc) or if some have custom values, which is why I don't display them correctly in the UI yet.

**Reply by ghramsey** [reply]
This character_class_combination seems to be very important information.
As it relates to my other question about aliens' triggering or not triggering the transit office door the combination max seems to be 1023.
The default on that door package is 733 and prevents aliens trigering the door, but allows droids.    I set 1023 and it allows aliens too.
So 1023 seems to be the index "ALL", but the values in between are still elusive.    

I wonder if I can back track through looking for the doors the NPC civvies use in M10 when told to take cover?
There are two one in Sys/Tec lobby and one in tech support.  If I can figure out what NPC only is and I know what
"ALL" is maybe I can figure out the amounts of each index step.   I'll start investigating that.  This is probably
only a piece of the puzzle since the NvMesh path is still probably in play but I feel that if the aliens are prohibited 
from waiting for the door at all they might automatically follow 'normal' procedure and up vent.

**Comment by ghramsey** [comment]
I had a feeling the enum values were incremental like that.   Figuring out the increment seems like it's going to be a headache.
Highest number I see though is 1023.  Maybe work backwards from that.   I'm out of my depth though. 

The only thing I've ever done with with numbers like that in programming was on a Commodore 64 and 8x8 sprite matrices in BASIC.
1-2-4-8-16-32-64-128 = 255 and tedious graph paper math for each row / column.   It's been over 30 yrs.

**Comment by ghramsey** [comment]
You've implemented this in the commands editor.  So this is now moot

**Reply by MattFiler** [reply]
This enum type isn't in yet - I have all the values but need to implement a UI for selecting the character combinations. Will be coming very soon to staging :)

**Reply by ghramsey** [reply]
OK.  IT sure seems like it's mostly done.
